webpack.config.js

```diff
@@ -1,31 +1,53 @@
-import path from 'path';
-import NodePolyfillPlugin from "node-polyfill-webpack-plugin";
-import CopyPlugin from "copy-webpack-plugin";
-import { fileURLToPath } from 'url';
+import HtmlWebpackPlugin from "html-webpack-plugin";
+import path from "path";
+import { fileURLToPath } from "url";
 
 const __filename = fileURLToPath(import.meta.url);
 const __dirname = path.dirname(__filename);
 
 export default {
-    mode: "production",
-    entry: {
-        index: path.resolve(__dirname, './src/js/index.js')
-    },
-    output: {
-        path: path.resolve(__dirname, 'dist'),
-    },
-    performance: {
-        maxAssetSize: 1000000,
-        maxEntrypointSize: 1000000,
-    },
-    plugins: [
-        new NodePolyfillPlugin(),
-        new CopyPlugin({
-            patterns: [
-                { from: "src/index.html" },
-                { from: "src/style.css" },
-            ],
-        }),
+  mode: "production",
+  entry: "./src/index.js",
+  performance: {
+    hints: false,
+    maxEntrypointSize: 512000,
+    maxAssetSize: 512000,
+  },
+  output: {
+    path: path.resolve(__dirname, "public"),
+
+    filename: "main.js",
+  },
+  devServer: {
```

Comment:
I prefer to have dev-related configs in a separate config like `webpack.dev.config.js`. Then the mode can also be set "development".